### ISSUE TITLE
Promotion property must not be redeclared

### ISSUE THREAD

**Issue by frkinta** [DEVSENSE/phptools-docs#257]
The code below may looks normal 

```PHP
 <?php

 class Test {
 
    private array $postedData;

    /**
     * 
     * @param array $postedData
     */
    public function __construct(
        private array $postedData,
    )
    {
    }

}
```
but it produce an error:  "Cannot redeclare Test::$postedData on line ... in file Test.php".
[see](https://stitcher.io/blog/constructor-promotion-in-php-8#no-duplicates-allowed)

Will be great to let know one is redeclaring.

Thanks

**Comment by jakubmisek** [comment]
thank you for the feature idea! working on it

**Comment by jakubmisek** [comment]
implemented - will be available in the next release (v1.30) or upcoming pre-release
